php/libraries/User.class.inc

```diff
@@ -182,13 +182,13 @@ class User extends UserPermissions
     /**
      * Gets user data
      *
-     * @param string $var Name of variable to get
+     * @param string|null $var Name of variable to get
      *
      * @note   Call without any parameters to get the entire user data array
      * @return mixed
      * @access public
      */
-    function getData($var = null)
+    function getData(?string $var = null)
```

Comment:
Why this function doesn't have a return type?

Reply:
@kongtiaowang 
Because it is *mixed* and the declaration do not support mixed return type (yet).

Reply:
Do you need the `?` for type hinting function parameters? I was under the impression that was only for return type declarations.

Reply:
@HenriRabalais I'm not actually sure about that... I don't know whether the Factory should accept null as a parameter even though the PHPDOC says that it should

Reply:
`?type` means "null or type" while `type` means "must be type and only type"